### PR TITLE
feat: Handle authentication without another activity

### LINKED ISSUE #736: issue with action bar title
**Describe the bug**
action bar title is not displaying the right content when navigating in the app. I think the issue arises because the logic behind changing the action bar title is given in the fragment that is being removed an instead in the fragment that is being created

**To Reproduce**

1.go to likes pannel
2.click on any event
3.come back to likes pannel by pressing back button
4.title will display Events instead of likes
5.click on likes pannel option in the bottom menu and it will start to display likes again

**Logs**

**Screenshots**

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |[redmi note 4]|
|Android Version|<[7.0]    |

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### LINKED ISSUE #777: Handle intent from attendee in sign up fragment
**Describe the feature you'd like**
Currently, attendee registration continues after sign-in authentication.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### LINKED ISSUE #793: Handle authentication in better way
**Describe the feature you'd like**
Authentication should call with start activity for results and get the result after authentication instead of clear all activity from back stack and start a new activity.

**It contains the following advantages** 
- No need to use flags for Activity clear top and set up the activity and fragment again, it contains bugs.
- No need to pass bundle to authentication activity -> Activity -> Fragment and get the same bundle in the same fragment after authentication.
- Only need to set results accordingly in auth activity and finish it.

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### LINKED ISSUE #848: Login Fragment gets opened on rotation in SIgnUpFragment
**Describe the bug**

On rotation of the device in sign up fragment, login fragment gets created. 

**To Reproduce**

1. Go to the sign up fragment .
2. Rotate the device.

**Expected behavior**

Sign up fragment should be shown.

**Logs**

**Screenshots**

sign_up

**Smartphone Info:**

|               | |
|---------------|-|
|Device         |One plus 5|
|Android Version|Android Pie 9.0.0 |

**Additional context**

**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other: 
@iamareebjamal @nikit19 Your views?

### PR DESCRIPTION
**Pull request by liveHarshit**
Part of #822 
Fixes #793 
Fixes #777 
Fixes #736 (Solved already with navigation architecture)
Fixes #848

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 1]
```diff
@@ -53,5 +57,22 @@ class MainActivity : AppCompatActivity() {
...
+                else -> View.GONE
+            }
+    }
+
+    override fun onBackPressed() {
+        when (currentFragmentId) {
+            R.id.navigation_login,
+            R.id.navigation_sign_up -> {
+                navController.popBackStack(R.id.navigation_events, false)
+                Toast.makeText(this@MainActivity, "Sign In canceled!", Toast.LENGTH_SHORT).show()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Snackbar

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt [hunk 2]
```diff
@@ -2,45 +2,49 @@ package org.fossasia.openevent.general
...
         setContentView(R.layout.activity_main)
 
         val hostFragment = supportFragmentManager.findFragmentById(R.id.frameContainer)
-        var navController: NavController? = null
-        if (hostFragment is NavHostFragment) {
+        if (hostFragment is NavHostFragment)
             navController = hostFragment.navController
-            setupBottomNavigationMenu(navController)
-        }
+        setupBottomNavigationMenu(navController)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: What if navController is not initialized?

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt [hunk 3]
```diff
@@ -120,28 +122,29 @@ class LoginFragment : Fragment() {
...
     private fun onEmailEntered(enable: Boolean) {
         rootView.loginButton.isEnabled = enable
         rootView.forgotPassword.isVisible = enable
     }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                findNavController(rootView).popBackStack(R.id.navigation_events, false)
+                Toast.makeText(context, "Sign In canceled!", Toast.LENGTH_SHORT).show()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Snackbar

**Comment by iamareebjamal** [comment]
There are UI changes, please add screenshots

**Comment by liveHarshit** [comment]
> There are UI changes, please add screenshots

UI is exactly the same as before. Earlier gray background is set in auth activity.

File: app/src/main/res/navigation/navigation_graph.xml [hunk 4]
```diff
@@ -59,4 +59,12 @@
         android:id="@+id/orderCompletedFragment"
         android:name="org.fossasia.openevent.general.order.OrderCompletedFragment"
         android:label="OrderCompletedFragment"/>
+    <fragment
+        android:id="@+id/navigation_login"
```
Comment (nikit19) [inline app/src/main/res/navigation/navigation_graph.xml]: Should be cameCase

Comment (liveHarshit) [reply app/src/main/res/navigation/navigation_graph.xml]: Yes, but it should be same with menu item ids for bottom navigation. Need to change all menu item ids for this. Should I change all menu item ids for this?

Comment (nikit19) [reply app/src/main/res/navigation/navigation_graph.xml]: All ids that are referenced in code should be in camelCase

Comment (liveHarshit) [reply app/src/main/res/navigation/navigation_graph.xml]: Updated

File: app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt (hunk 3 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/auth/LoginFragment.kt]: Updated

File: app/src/main/java/org/fossasia/openevent/general/MainActivity.kt (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/MainActivity.kt]: Second part

**Pull request merged**